### PR TITLE
fix @-isdefined macro for static-parameters

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: src/interpreter.c [hunk 1]
```diff
@@ -237,6 +240,18 @@ static jl_value_t *eval(jl_value_t *e, interpreter_state *s)
         else if (jl_is_symbol(sym)) {
             defined = jl_boundp(modu, (jl_sym_t*)sym);
         }
+        else if (jl_is_expr(sym) && ((jl_expr_t*)sym)->head == static_parameter_sym) {
+            ssize_t n = jl_unbox_long(args[0]);
+            assert(n > 0);
+            if (s->sparam_vals && n <= jl_svec_len(s->sparam_vals)) {
+                jl_value_t *sp = jl_svecref(s->sparam_vals, n - 1);
+                defined = !jl_is_typevar(sp);
```
Comment (JeffBezanson) [inline src/interpreter.c]: I assume `jl_is_typevar` should never be true now, but will be in the near future?

Comment (vtjnash) [inline src/interpreter.c]: yeah

File: src/codegen.cpp [hunk 2]
```diff
@@ -3112,6 +3112,12 @@ static jl_cgval_t emit_isdefined(jl_codectx_t &ctx, jl_value_t *sym)
             }
         }
     }
+    else if (jl_is_expr(sym)) {
+        assert(((jl_expr_t*)sym)->head == static_parameter_sym && "malformed isdefined expression");
+        size_t i = jl_unbox_long(jl_exprarg(sym, 0)) - 1;
+        (void)i;
```
Comment (tkelman) [inline src/codegen.cpp]: why give this output a variable name at all?

Comment (vtjnash) [inline src/codegen.cpp]: this is actually a backport from another PR that'll need the variable

**Pull request merged**